src/test/fuzz/process_messages.cpp

```diff
@@ -48,14 +48,16 @@ void test_one_input(const std::vector<uint8_t>& buffer)
     const auto num_peers_to_add = fuzzed_data_provider.ConsumeIntegralInRange(1, 3);
     for (int i = 0; i < num_peers_to_add; ++i) {
         const ServiceFlags service_flags = ServiceFlags(fuzzed_data_provider.ConsumeIntegral<uint64_t>());
-        const ConnectionType conn_type = fuzzed_data_provider.PickValueInArray({ConnectionType::INBOUND, ConnectionType::OUTBOUND_FULL_RELAY, ConnectionType::MANUAL, ConnectionType::FEELER, ConnectionType::BLOCK_RELAY, ConnectionType::ADDR_FETCH});
+        const NetPermissionFlags permission_flags = NetPermissionFlags(fuzzed_data_provider.ConsumeIntegral<std::underlying_type<NetPermissionFlags>::type>());
+        const ConnectionType conn_type = fuzzed_data_provider.PickValueInArray(ALL_CONNECTION_TYPES);
         peers.push_back(MakeUnique<CNode>(i, service_flags, 0, INVALID_SOCKET, CAddress{CService{in_addr{0x0100007f}, 7777}, NODE_NETWORK}, 0, 0, CAddress{}, std::string{}, conn_type).release());
         CNode& p2p_node = *peers.back();
 
+        connman.SetPeerPermissionFlags(p2p_node, permission_flags);
         p2p_node.fSuccessfullyConnected = true;
-        p2p_node.fPauseSend = false;
         p2p_node.nVersion = PROTOCOL_VERSION;
         p2p_node.SetCommonVersion(PROTOCOL_VERSION);
+        p2p_node.fPauseSend = false;
```

Comment:
Why was this moved?

Reply:
There are some more changes that I haven't yet pushed to increase coverage even further. They change the seed format as well, so I am going to mark this as draft for now.

Reply:
What is the procedure for when a PR needs to modify the fuzzing harness and may invalidate all of the seeds? Should something akin to a database migration be used if possible, applied to the seeds to get at least the same coverage as before?

Reply:
The format might be arbitrarily complex (even depending on the consensus rules or other Bitcoin Core code), so I don't think this can be done automatically in a trivial manner. I mostly do it manually if the fuzz engine can't figure it out by itself after a bit of CPU time.